src/main/app/case/CaseApi.ts

```diff
@@ -71,13 +72,26 @@ export class CaseApi {
     }
   }
 
-  public async triggerEvent(caseId: string, caseData: Partial<Case>, eventName: string): Promise<void> {
+  public async triggerEvent(caseId: string, userData: Partial<Case>, eventName: string): Promise<CaseWithId> {
     const tokenResponse = await this.axios.get(`/cases/${caseId}/event-triggers/${eventName}`);
     const event = { id: eventName };
-    const data = toApiFormat(caseData);
+    const data = toApiFormat(userData);
 
     try {
-      await this.axios.post(`/cases/${caseId}/events`, { event, data, event_token: tokenResponse.data.token });
+      const response = await this.axios.post(`/cases/${caseId}/events`, {
+        event,
+        data,
+        event_token: tokenResponse.data.token,
+      });
+      const caseData = { id: response.data.id, ...fromApiFormat(response.data.data) };
+
+      for (const [key, value] of Object.entries(userData)) {
+        if (!isEqual(caseData[key], value ?? '')) {
```

Comment:
can this be handled by the fromApiFormat function?

Reply:
👍 `value` was incorrectly being set to `undefined` when a checkbox was unchecked, I've fixed this in 9228f0a.